solidity/contracts/bridge/VendingMachine.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../token/TBTCToken.sol";
+import "../token/IReceiveApproval.sol";
+import "../GovernanceUtils.sol";
+
+/// @title TBTC v2 Vending Machine
+/// @notice The Vending Machine is the owner of TBTC v2 token and can mint
+///         TBTC v2 tokens in 1:1 ratio from TBTC v1 tokens. TBTC v2 can be
+///         unminted back to TBTC v1 with or without a fee - fee parameter is
+///         controlled by the governance. This implementation acts as a bridge
+///         between TBTC v1 and TBTC v2 token, allowing to mint TBTC v2 before
+///         the system is ready and fully operational without sacrificing any
+///         security guarantees and decentralization of the project.
+///         Vending Machine as a contract itself is not upgradeable, though
+///         TBTC v2 token ownership can be updated in a two-step,
+///         governance-controlled process. It is expected that this process
+///         will be executed before the v2 system launch.
+contract VendingMachine is Ownable, IReceiveApproval {
+    using SafeERC20 for IERC20;
+    using SafeERC20 for TBTCToken;
+
+    /// @notice The time delay that needs to pass between initializing and
+    ///         finalizing update of any governable parameter in this contract.
+    uint256 public constant GOVERNANCE_DELAY = 48 hours;
```

Comment:
What would the panic button do?

Reply:
Block the possibility of minting new tokens.

Reply:
Which would probably be achievable indirectly by changing the owner of the TBTCv2 contract.

Reply:
Yes, if the DAO decides minting new tokens should be paused, they need to upgrade to the new vending machine with paused minting. I expect it might happen shortly before v2 system launch so that v1 can be safely shut down in a coordinated redemption churning.